src/controller/java/src/chip/devicecontroller/AndroidChipStack.java

```diff
@@ -65,6 +65,98 @@
     }
   }
 
+  public static class ChipGattCallback extends BluetoothGattCallback {
+    @Override
+    public void onConnectionStateChange(BluetoothGatt gatt, int status, int newState) {
+      if (newState == BluetoothProfile.STATE_DISCONNECTED) {
```

Comment:
nit: you could decrease indent by switching condition and early-returning here and below:

```
if (newState != BluetoohProfile.STATE_DISCONNECTED) {
  return;
}

int connId = getInstance().getConnId(gatt);
...
```

Reply:
right, I'll change it